internal/interfaces/http/tdex_connect_handler.go

```diff
@@ -32,31 +35,42 @@ type tdexConnect struct {
 	walletUnlockerSvc application.WalletUnlockerService
 	macaroonBytes     []byte
 	certBytes         []byte
-	serverPort        string
 	macaroonPath      string
 	certPath          string
+	scheme            string
+	serverAddress     string
 }
 
 func NewTdexConnectService(
 	repoManager ports.RepoManager,
 	walletUnlockerSvc application.WalletUnlockerService,
-	macaroonPath, certPath, serverPort string,
-) TdexConnectService {
+	macaroonPath, certPath, serverPort, host, protocol string,
+) (TdexConnectService, error) {
 	macBytes := readFile(macaroonPath)
 	certBytes := readFile(certPath)
 
+	scheme := protocol
+	if certBytes != nil {
+		if protocol == "" || protocol == httpsProtocol {
+			scheme = httpsProtocol
+		} else if protocol == httpProtocol {
+			return nil, errors.New("http protocol invalid with cert provided")
+		}
+	}
```

Comment:
we should move this validation logic to the encoding function of `pkg/tdexdconnect` to avoid rewriting the same code where else that package is used and also to enforce that it generates only correct URLs.

Reply:
Maybe we need also a Validate function of sort? so we stop before trying to encoding the string

Reply:
Done